LogAnalyzer.groovy

```diff
@@ -0,0 +1,136 @@
+        this.handler = pHandler
+        initialize_maps()
+    }
+
+
+    void calculate_maps(){
+        println(this.handler.get_old_logs().size())
+        calculate_common_queries()
+        calculate_common_filters()
+        calculate_common_queries_with_filters()
+        calculate_common_requests()
+    }
+
+
+    // ---------------------------------------------------------------------------------------------------
+    // Description :
+    //      The following calculate the times an entry has been seen in the parsed new and old log formats
+    // ---------------------------------------------------------------------------------------------------
+    void calculate_common_queries() {
+        this.handler.get_old_logs().each { log ->
+            add_occurrence(this.commonQueries, log.get_search_query())
+        }
+        this.handler.get_search_logs().each { log ->
+            add_occurrence(this.commonQueries, log.get_search_query())
+        }
+        this.commonQueries = this.commonQueries.sort { -it.value }
+    }
+
+
+    void calculate_common_filters() {
+        this.handler.get_old_logs().each { log ->
+            log.get_filters().each { filter ->
+                add_occurrence(this.commonFilters, filter.asType(String))
+            }
+        }
+        this.handler.get_search_logs().each { log ->
+            log.get_filters().each { filter ->
+                add_occurrence(this.commonFilters, filter.asType(String))
+            }
+        }
+        this.commonFilters = this.commonFilters.sort { -it.value }
+
+    }
+
+
+    void calculate_common_queries_with_filters(){
+        this.handler.get_old_logs().each { log ->
+            String queryAndFilter = "${log.get_search_query()} ${log.get_filters()}"
+            add_occurrence(this.commonQueriesAndFilters, queryAndFilter)
+        }
+        this.handler.get_search_logs().each { log ->
+            String queryAndFilter = "${log.get_search_query()} ${log.get_filters()}"
+            add_occurrence(this.commonQueriesAndFilters, queryAndFilter)
+        }
+        this.commonQueriesAndFilters = this.commonQueriesAndFilters.sort { -it.value }
+    }
+
+
+    void calculate_common_requests() {
+        this.handler.get_request_logs().each { log ->
+            add_occurrence(this.commonRequests, log.get_id())
+        }
+        this.commonRequests = this.commonRequests.sort { -it.value }
+    }
+
+
+    // Getters
+    // -------
+    Map<String, Integer> get_common_queries(){
+        return this.commonQueries
+    }
+
+
+    Map<String, Integer> get_common_filters(){
+        return this.commonFilters
+    }
+
+
+    Map<String, Integer> get_common_query_filter(){
+        return this.commonQueriesAndFilters
+    }
+
+
+    Map<String, Integer> get_common_requests(){
+        return this.commonRequests
+    }
+
+
+
+
+
+    // ------------------------
+    // PRIVATE DATA AND METHODS
+    // ------------------------
```

Comment:
Setting the fields here privately is good because we don't want the setters. This is preference, but because we only have one constructor (actually there are two because of Groovy), but it probably is better to declare the fields at the top, and either bring in initialize_maps into the constructor (get rid of the private method) or move the private method to the top. I would say it is not necessary to have another method because we only are using it one constructor; my main point here is that handling the fields should happen close to where the fields are declared, which is above the constructor.